challenges.js

```diff
@@ -4,81 +4,79 @@ class Challenge {
 
-    const img = document.createElement('img');
-    img.classList.add('challenges-container__challenge__img');
-    img.src = this.data.image + '?image=' + Math.floor(Math.random() * 16);
+    const img = document.createElement("img");
+    img.classList.add("challenges-container__challenge__img");
+    img.src = this.data.image + "?image=" + Math.floor(Math.random() * 16);
     challengeCard.append(img);
 
-    const titleDiv = document.createElement('div');
-    titleDiv.classList.add('challenges-container__challenge__lowerWrapper');
+    const titleDiv = document.createElement("div");
+    titleDiv.classList.add("challenges-container__challenge__lowerWrapper");
     challengeCard.append(titleDiv);
 
-    const title = document.createElement('h3');
-    title.classList.add('.challenges-container__challenge__title');
+    const title = document.createElement("h3");
+    title.classList.add(".challenges-container__challenge__title");
     title.textContent = this.data.title;
     titleDiv.append(title);
 
-    const type = document.createElement('h3');
-    type.classList.add('challenges-container__challenge__type');
-    type.textContent = '(' + this.data.type + ')';
+    const type = document.createElement("h3");
+    type.classList.add("challenges-container__challenge__type");
+    type.textContent = "(" + this.data.type + ")";
     titleDiv.append(type);
 
-    const ratingContainer = document.createElement('small');
-    ratingContainer.classList.add('challenges-container__challenge__rating');
+    const ratingContainer = document.createElement("small");
+    ratingContainer.classList.add("challenges-container__challenge__rating");
     challengeCard.append(ratingContainer);
 
-    const rating = document.createElement('span');
-    rating.classList.add('challenges-container__challenge__rating__stars');
-    rating.ariaLabel = 'Rating';
-    rating.role = 'meter';
-    rating.ariaValueMin = '0';
-    rating.ariaValueMax = '5';
-    rating.ariaValueNow = Math.ceil(this.data.rating);
+    const rating = document.createElement("span");
+    rating.classList.add("challenges-container__challenge__rating__stars");
+    rating.ariaLabel = "Rating";
+    rating.role = "meter";
+    rating.ariaValueMin = "0";
+    rating.ariaValueMax = "5";
+    rating.ariaValueNow = this.data.rating;
     ratingContainer.append(rating);
 
-    const star1 = document.createElement('i');
-    star1.ariaHidden = true;
-    rating.append(star1);
-
-    const star2 = document.createElement('i');
-    star2.ariaHidden = true;
-    rating.append(star2);
-
-    const star3 = document.createElement('i');
-    star3.ariaHidden = true;
-    rating.append(star3);
-
-    const star4 = document.createElement('i');
-    star4.ariaHidden = true;
-    rating.append(star4);
-
-    const star5 = document.createElement('i');
-    star5.ariaHidden = true;
-    rating.append(star5);
+    function addStar(starRating, starNumber) {
```

Comment:
Nice and simple solution! I do find the naming of the parameters a little bit confusing though, and I will try to explain why in a way that can be useful for the future.

Variable naming (like all communication) always happen within a communicative context. I think that your name relies on the context being something different than what it is here.

If someone asks you "What is the star rating of this challenge?" (i.e. in the context of a challenge), it's quite obvious that they are intending that question in a way that is synonymous with "How many stars does this challenge have?". So in that situation, "star rating" means "number of stars".

But the context here is not a challenge. The name of the function is `addStar()`, so the context here is a single star. This function exists to add a single star (and can then be used in a loop to add multiple stars), so the context, in my opinion, is a single star.

In that context, I interpret the variable name `starRating` as having the same meaning that you may intend if you ask a question like "What is the rating (or value) of this star?". But it's the second parameter, `starNumber` that indicates the value of the star we're creating.

So I think `starRating` would have been better named something like `challengeRating` or `totalRating`, to indicate in this context (where we've zoomed in on a single star) that it's not related to this particular star, but related to something different (or more zoomed out).